### ISSUE TITLE
refs --reveal --ref-file fails (at least with GKMS)

### ISSUE THREAD

**Issue by drzero42** [deepmind/kapitan#604]
**Describe the bug/feature**
The new feature in 0.29.0 which adds `--ref-file` to the `kapitan refs` command does not seem to work correctly.
If I use `kapitan refs --reveal --file compiled/targetname/file.yaml` it works correctly.
The same goes for `kapitan refs --reveal --tag "?{gkms:targets/targetname/mysecret}".
However, when I then try `kapitan refs --reveal --ref-file refs/targets/targetname/mysecret` it gives me one of these two outputs:
```
Traceback (most recent call last):
  File "kapitan/__main__.py", line 10, in <module>
  File "kapitan/cli.py", line 551, in main
  File "kapitan/refs/cmd_parser.py", line 28, in handle_refs_command
  File "kapitan/refs/cmd_parser.py", line 282, in ref_reveal
  File "kapitan/refs/secrets/gkms.py", line 87, in reveal
  File "base64.py", line 87, in b64decode
binascii.Error: Incorrect padding
[211851] Failed to execute script __main__
```
or
```
Traceback (most recent call last):
  File "kapitan/refs/secrets/gkms.py", line 150, in _decrypt
  File "googleapiclient/_helpers.py", line 130, in positional_wrapper
  File "googleapiclient/http.py", line 856, in execute
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://cloudkms.googleapis.com/v1/projects/XXX/locations/global/keyRings/XXX/cryptoKeys/XXX:decrypt?alt=json returned "Decryption failed: the ciphertext is invalid.">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "kapitan/__main__.py", line 10, in <module>
  File "kapitan/cli.py", line 551, in main
  File "kapitan/refs/cmd_parser.py", line 28, in handle_refs_command
  File "kapitan/refs/cmd_parser.py", line 282, in ref_reveal
  File "kapitan/refs/secrets/gkms.py", line 88, in reveal
  File "kapitan/refs/secrets/gkms.py", line 156, in _decrypt
kapitan.refs.secrets.gkms.GoogleKMSError: <HttpError 400 when requesting https://cloudkms.googleapis.com/v1/projects/XXX/locations/global/keyRings/XXX/cryptoKeys/XXX:decrypt?alt=json returned "Decryption failed: the ciphertext is invalid.">
[211934] Failed to execute script __main__
```

**To Reproduce**
1. Add some kapitan secrets using GKMS
2. Run `kapitan refs --reveal --ref-file path/to/secret`

**Expected behavior**
I expected the decrypted version of the secret to be output.

** If it's a bug (please complete the following information):**
 - `python --version`: Python 3.8.2
 - `pip3 --version`: pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)
 - `Are you using pyenv or virtualenv?` Neither, am using single-binary downloaded from Github release

**Comment by ramaro** [comment]
Thanks for reporting! Will be picking this one up...

**Comment by AlphaRoy14** [comment]
Hi @ramaro, I think this is happening because of `encrypt=True` here https://github.com/deepmind/kapitan/blob/c2f13f27b0172c081f4c7561d53e737cb0047699/kapitan/refs/base.py#L522 I have updated this by setting `encrypt=False`in my Azure KMS backend PR, if you like I can make a separate PR for this bug?

**Comment by ramaro** [comment]
@AlphaRoy14 Yes good point, that is most likely it! The comment mentions that but the code is setting it wrong :) This also means we need to add some tests for encrypted refs on this feature (likely using GPG).

**Comment by AlphaRoy14** [comment]
I'll add a test case for it :)

**Comment by ramaro** [comment]
Thanks, something like https://github.com/deepmind/kapitan/blob/master/tests/test_cli.py#L396 but for GPG should do!

**Comment by uberspot** [comment]
Thanks for reporting and thanks @AlphaRoy14 for the fix. 
This will be included in the 0.29.1 release started here: https://github.com/deepmind/kapitan/pull/606
Closing issue for now. :+1:
